File: Python/5kyu/Tic-Tac-Toe Checker.py
# https://www.codewars.com/kata/525caa5c1bf619d28c000335/

def is_solved(board):
    # Check rows
    for row in board:
        if len(set(row)) == 1 and row[0] != 0:
            return row[0]
    # Check columns
    for i in range(3):
        if len(set([row[i] for row in board])) == 1 and board[0][i] != 0:
            return board[0][i]
    # Check diagonals
    if len(set([board[i][i] for i in range(3)])) == 1 and board[0][0] != 0:
        return board[0][0]
    if len(set([board[i][2-i] for i in range(3)])) == 1 and board[0][2] != 0:
        return board[0][2]
    # Check if there are any zeros
    for row in board:
        if 0 in row:
            return -1
    # If no zeros and no winner, it's a draw
    return 0

File: JavaScript/5kyu/Tic-Tac-Toe Checker.js
// https://www.codewars.com/kata/525caa5c1bf619d28c000335/

function isSolved(board) {
    const win = (arr, n) => arr.every(x => x === n);
    const checkDiag = (arr, n) => arr[0][0] === n && arr[1][1] === n && arr[2][2] === n || arr[0][2] === n && arr[1][1] === n && arr[2][0] === n;
    const checkRow = (arr, n) => arr.some(x => win(x, n));
    const checkCol = (arr, n) => checkRow(arr[0].map((_, c) => arr.map(r => r[c])), n);
    const checkAll = (arr, n) => checkRow(arr, n) || checkCol(arr, n) || checkDiag(arr, n);
    const checkDraw = (arr) => arr.every(x => x.every(y => y !== 0));
    if (checkAll(board, 1)) return 1;
    if (checkAll(board, 2)) return 2;
    if (checkDraw(board)) return 0;
    return -1;
  }